### ISSUE TITLE
Emptied milk buckets not being taken from Baker's storage when done with the Milk

### ISSUE THREAD

**Issue by Knaxia** [ldtteam/minecolonies#6035]
### Prerequisites 

- [x] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [x] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version: 1.15.2
- MineColonies Version: 0.13.415-ALPHA
- Structurize Version: 0.13.94-ALPHA
- Related Mods and their Versions: Everything from the MineColonies Official pack By: H3lay, Kostronor, D3miurge_, Wissin8r, Asherslab on Twitch (updated on the 22nd of October, 2020)

## Expected behavior
Once the baker is done making something that requires milk, the empty buckets are stored, then taken by a courier from their hut and put into the warehouse storage. Which are then delivered to the Cowboy to get more milk.

## Actual behavior
Once the baker has completed with the milk, the empty buckets are stored in the Baker's hut storage (main storage or racks) and stay there. No couriers ever go grab them. Even if I call in a courier to come pick up stuff, the buckets will stay in there. I have to manually grab them and drop them into a stash or give them to the cowboy manually.

## Steps to reproduce the issue

1. Have an active warehouse and courier
2. Have a working cowboy with cows, enabled with "collect Milk" and make sure he has 3 buckets and wait for him to fill them with milk and that they are all put into storage at the warehouse
3. Have a level 2 baker, which knows how to make a cake 
4. Have a mailbox and request a cake and wait for it to be crafted and delivered.
5. Wait OR request buckets from the mailbox OR wait for the cowboy to request buckets.

The buckets will stay in the bakery storage unless you manually go there and request a pickup. Otherwise, they will never go and grab it. Even if set to 10/10 priority, it doesn't seem to be an item that is on the list of "needs to grab when unused". Probably because of the cowboy that also requires empty ones.

### Logs

None needed, no errors related to them in there.

### Notes
I've waited a little for many updates just to make sure, I've started my colony on 0.13.271-ALPHA and ever since I built the bakery, I've always had that problem. In the past, my buckets actually vanished, I couldn't find them anywhere in storage of all my colony and I checked every building / possible storage spot. I made 9 more (enough for 3 cakes per request), they no longer just vanish, but the problem of them staying stuck at the bakery persists.

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by Knaxia** [comment]
I have updated the steps as requesting a pickup from a Courier will actually grab the buckets now (wasn't before). The issue still remains that they will never automatically go and take the empty buckets back.

I have to add that behavior that MIGHT also be affected by the same issue:

Items which are classified as "Usable" (weapons, armor and tools) will always stick in the inventory of people or  workplace storage, even if they don't require them or need to use them.

I've seen:
-Guard towers with random extra armor pieces in them, swords or even axes, even if the guard assigned to that tower is fully geared
-Guards running around with their armor + weapon assigned to their Knight/Archer status, but randomly have other weapons on them at all time, even if it's not their use. Knights with bows and archers with swords. They are probably from fight on barbarians and such , but they always keep just one on themselves and put the rest in their designed tower for pickup.

**Comment by Raycoms** [comment]
Yes. guards on purpose store another set in case their current one breaks mid battle. That is intended

**Comment by Knaxia** [comment]
> 
> 
> Yes. guards on purpose store another set in case their current one breaks mid battle. That is intended

But even if it's something they don't use? Like a bow on a Knight or an axe on an archer?

**Comment by ravenbuilder934** [comment]
I've even seen knights carrying bows in their inventory

**Comment by Mekle001** [comment]
I've got a change that fixes milk buckets I'm working on, if I can find the chance to finish debugging it.
